docs2/site/docs/migrations/migration4.md

```diff
@@ -68,6 +68,32 @@ requested type via `ObjectExtensions.ToObject` as it did before.
 
 (sungam3r todo: write separate page on these features, write a simple summary and reasoning here, and add link to new page)
 
+### Microsoft-specific Dependency Injection Extensions
+
+If you are using the Microsoft dependency injection library, extension methods are provided within
+the [GraphQL.MicrosoftDI NuGet package](https://www.nuget.org/packages/GraphQL.MicrosoftDI) for creating a service provider scope
+during a field resolver's execution. This is useful when accessing a scoped service with a parallel execution strategy, as
+typically scoped services are not multi-threaded compatible. The library also provides a builder to assist constructing
+a field resolver that relies on scoped services. Below is a sample of a field resolver that relies on a scoped
+service and can run concurrently with other field resolvers:
+
+```csharp
+public class MyGraphType : ObjectGraphType<Category>
+{
+    public MyGraphType()
+    {
+        Field("Name", context => context.Source.Name);
+        Field<ListGraphType<ProductGraphType>>().Name("Products")
+            .Resolve()
```

Comment:
Two `Resolve`/`ResolveAsync` look strange.

Reply:
I copied the pattern used by linq2db for their merge builder.  I think it works really well...

Reply:
What I'm saying is that when I look at the code, I see two `Resolve` calls - one without parameters, the second with parameters. And a few more lines of code in between. It's not obvious.

Reply:
Maybe I'm used to it.  It works really well in the linq2db library.  I'm on their team and I've not seen questions about it.  Of course everything needs documentation, but it lays out real well.

Reply:
Take a look at their merge API - https://linq2db.github.io/articles/sql/merge/Merge-API.html

Notice that it starts with `Merge()` and ends with either `Merge()` or `MergeAsync()`.

Reply:
At least `Resolve` -> `ConfigureResolve`

Reply:
We do not use such a convention for field builders or connection builders

Reply:
Yes but here one method starts to conflict with another method in its meaning. They just overlap. It's not about conventions, it's about meaning.

Reply:
ok